extensions/amp-form/0.1/amp-form.js

```diff
@@ -242,22 +247,29 @@ export class AmpForm {
           xhrUrl = this.xhrAction_;
           body = new FormData(this.form_);
         }
-        return this.xhr_.fetchJson(dev().assertString(xhrUrl), {
+        return this.xhr_.fetch(dev().assertString(xhrUrl), {
           body,
           method: this.method_,
           credentials: 'include',
         }).then(response => {
-          this.triggerAction_(/* success */ true, response);
-          // TODO(mkhatib, #6032): Update docs to reflect analytics events.
-          this.analyticsEvent_('amp-form-submit-success');
-          this.setState_(FormState_.SUBMIT_SUCCESS);
-          this.renderTemplate_(response || {});
-        }).catch(error => {
+          return response.json().then(json => {
+            this.triggerAction_(/* success */ true, json);
+            this.analyticsEvent_('amp-form-submit-success');
+            this.setState_(FormState_.SUBMIT_SUCCESS);
+            this.renderTemplate_(json || {});
+            this.maybeHandleRedirect_(
+                /** @type {../../../src/service/xhr-impl.FetchResponse} */ (
+                    response));
+          });
+        }, error => {
```

Comment:
The reason I moved it here is that I didn't want it to catch any of that since that doesn't mean the submission has failed. 

But you're right we still need another catch to handle the JSON parsing errors. So I'd rather add it to the `json().then(jsonParsedSuccess, jsonFailed)` block instead. Reason being I don't want the re-catch and rethrow the error I am throwing inside `maybeHandleRedirect_`. 

Does that make sense?